extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/runtime/BeanUtils.java

```diff
@@ -0,0 +1,93 @@
+        if (mongoClient != null) {
+            return mongoClient;
+        }
+
+        if ((entity == null) || entity.clientName().isEmpty()) {
+            // this case happens when there are multiple instances because they are all annotated with @Named
+            for (InstanceHandle<T> handle : Arc.container().select(clientClass).handles()) {
+                InjectableBean<T> bean = handle.getBean();
+                boolean hasNamed = false;
+                for (Annotation qualifier : bean.getQualifiers()) {
+                    if (qualifier.annotationType().equals(Named.class)) {
+                        hasNamed = true;
+                    }
+                }
+                if (!hasNamed) {
+                    return handle.get();
+                }
+            }
+            throw new IllegalStateException(String.format("Unable to find default %s bean", clientClass.getSimpleName()));
+        } else {
+            throw new IllegalStateException(
+                    String.format("Unable to find %s bean for entity %s", clientClass.getSimpleName(), entity.toString()));
+        }
+    }
+
+    @SuppressWarnings("rawtypes")
+    private static AnnotationLiteral clientLiteral(String name) {
+        if (name.startsWith(MongoClientBeanUtil.DEFAULT_MONGOCLIENT_NAME)) {
+            return Default.Literal.INSTANCE;
+        }
+        return NamedLiteral.of(name);
+    }
+
+    public static String getDatabaseName(MongoEntity entity, String clientBeanName) {
+        MongoClients mongoClients = Arc.container().instance(MongoClients.class).get();
+        MongoClientConfig matchingMongoClientConfig = mongoClients.getMatchingMongoClientConfig(clientBeanName);
+        if (matchingMongoClientConfig.database.isPresent()) {
+            return matchingMongoClientConfig.database.get();
+        }
+
+        if (!clientBeanName.equals(MongoClientBeanUtil.DEFAULT_MONGOCLIENT_NAME)) {
+            MongoClientConfig defaultMongoClientConfig = mongoClients
+                    .getMatchingMongoClientConfig(MongoClientBeanUtil.DEFAULT_MONGOCLIENT_NAME);
+            if (defaultMongoClientConfig.database.isPresent()) {
+                return defaultMongoClientConfig.database.get();
+            }
+        }
+
+        if (entity == null) {
+            throw new IllegalArgumentException(
+                    "The database property was not configured for the default Mongo Client (via 'quarkus.mongodb.database'");
+        }
+        if (entity.clientName().isEmpty()) {
+            throw new IllegalArgumentException("The database method was not set for : " + entity
+                    + " and neither was the database property configured for the default Mongo Client (via 'quarkus.mongodb.database')");
+        }
+        throw new IllegalArgumentException(String.format(
+                "The database method was not set for : %s and neither was the database property configured for the named Mongo Client (via 'quarkus.mongodb.%s.database')",
```

Comment:
Same here I would rephrase to something like

```
The database attribute was not set for the @MongoEntity annotation and neither ...
```

Reply:
Done